js/preferences/Preferences.js

```diff
@@ -379,14 +562,20 @@ export async function __setup__() {
 
     }
     
-    prefs = Object.create(prefs_default);
-    Object.assign(prefs, loaded_prefs);
+    prefs = structuredClone(prefs_default);
+    deep_assign(prefs, loaded_prefs);
```

Comment:
I'm a bit curious why foregoing `Object.create` for `structuredClone` here, the goal was to fallback to the default values when none are specified.

Reply:
I did this because `JSON.stringify` does not stringify the prototypes of objects, like what happens when doing `Object.create`. This is fine for "top-level" properties, since those get assigned and copied over into the "actual" properties, but for properties inside other objects (which is the case for runtime options in the config file), the object in the prototype gets mutated instead and JSON.stringify passes over it, causing it to not be saved into the config file.
![image](https://github.com/thennothinghappened/GMEdit-Constructor/assets/68464103/f689f24d-bae8-440f-bdd7-0f7672dd878c)